----

## Reminders before submitting any phase 

### Please delete this section/these reminders BEFORE your final final submission at the end of the semester

* **TECHNIQUES FROM CLASS**:  Be sure to rely on your techniques from class when working on your project.  Points wil  be deducted if you skip over relevant technqiques that should  be considered.  Examples:
    * Correlation coefficents should be given for scatterplots;
    * If you give a p-value (which you should at some point), you should interpret it and also be sure to include the null and alternative hypotheses.
    * If you're doing a regression, be sure to desicuss the properties of the residual plot.
    * GridSearchCV should be used to help hypertune your parameters.
    * etc.
* **FLOW**: Eventually, you need to have FLOW in your project, where one section flows into the next.
    * So each section should have an opening sentence for readibility introducing the point of the section.
    * Also, at the end of the section, you must include a discussion of what question you plan to ask next or where your analysis is heading.
    * Your goal is to create a *READABLE*, professional writeup.
* **AUDIENCE**:   Your audience is a company, not an educator.  So pretend you are a consultant for a company and presenting results to them.  Be professional.  Include any background that the reader might need to know to understand the question.
* **GRAPHS**:  We love them.  Be sure to xplain what you learn from the graph AFTER the graph picture.
    * Each graph needs a title, y axis labels, and x axis labels, and legends if necessary.   
* **TABLE OF CONTENTS:**  You need an organized table of contents showing all your subsections and helping the reader read through your document.
* There should be **no crappy "text"** printed before each picture.
    * Add a semicolon to the last line int he code cell to prevent this.
* In each code cell, **there must be a code comment** explaining what is going on in the code cell.
    * NOOOOOOO results should be discussed in the hashtag areas
    * Example of a good code comment:  #In this code cell, we build a line graph of the number of reservations across time.
* If there is **anything special going on in your code cell, that should be documented and explained with code comments**  and possibly in the discussion before the code cell, if appropriate.
    * For example, if you are filtering on a special condition (like where tuition > 0 for some reason) then explain why.
* **Long Code:**  Coders don't like to scroll right!
    * If you have long code that extends to the right so that I'd have to scroll to see it, then just hit enter after a comma and the code will be placed indented on the next line for you.   
* **"Significant"** ->  Only use this word if you have done a hypothesis test... "There is a **significant** difference between the average launch angle in 2022 and in 2015."
    * If not using a hypothesis test/p-value, then use a synonym like  **striking**.
    * So know that the word "significant"  requires statistical evidence.  
* **Correlation does NOT imply causation** so don't accidentally use lingo that suggests that "variable X" is CAUSING some change in varaiable Y:
    * SHOULD NOT SAY:  
        * X is impacting Y
        * X causes Y
        * X contributes to Y
        * X influences to Y
        * X leads to Y
        * X increases Y
        * etc
    * Instead, just keep your lingo speculative and then you can make guesses:
      * It is possible that we see this effect because of XXX.
      * Perhaps this makes sense because XXX.

---

In [2]:
# imports go here
import os

#For reading the dataset
import kagglehub
from kagglehub import KaggleDatasetAdapter
import pandas as pd
import numpy as np

c:\Users\carlo\OneDrive\Escritorio\Universidad\2Cuatri\MachineLearning\FinalProject\Repo\FinalProjectCS-362\venv\Lib\site-packages\tqdm\auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


---
#### CS 362 - Machine Learning - Final Project - SP26

---

# Title (Give a title, based on your topic. Also be sure to rename the file with your initials.)

### By: Carlos Parra Camacho and Raymond Vargas Pérez


---

## 1. Introduction

For this project, our group explored... (This section will be finished at a later date and can be skipped at this time.)

When you do write it: I expect 2-3 paragraphs, more if needed.

---

## 2. Data

A direct link to the data can be found at the following website. 
https://www.kaggle.com/datasets/jboysen/mri-and-alzheimers

**Dr. Note to students - Fill in the details of how/when the data was obtained and delete this bolded comment. Example:** According the website, the data was collected via "telephone interviews conducted February 5-8, 2015 among a national sample of 1,003 adults, 18 years of age or older, living in the continental United States."   We believe the data is legitimate because XXX

**Dr. Note to students - Fill in the details of the code cell below for your project.  Be sure to read the comments in the cell for understanding, and then delete this comment.**

In [3]:
#In this cell, we load the data and rename columns appropriately
path = kagglehub.dataset_download("jboysen/mri-and-alzheimers")

# Load the CSV file directly using pandas
csv_file = os.path.join(path, "oasis_cross-sectional.csv")  # This is the main data file
mri_df = pd.read_csv(csv_file)

# Display the loaded data
mri_df.head()

,ID,M/F,Hand,Age,Educ,SES,MMSE,CDR,eTIV,nWBV,ASF,Delay
0,OAS1_0001_MR1,F,R,74,2.0,3.0,29.0,0.0,1344,0.743,1.306,NaN
1,OAS1_0002_MR1,F,R,55,4.0,1.0,29.0,0.0,1147,0.810,1.531,NaN
2,OAS1_0003_MR1,F,R,73,4.0,3.0,27.0,0.5,1454,0.708,1.207,NaN
3,OAS1_0004_MR1,M,R,28,NaN,NaN,NaN,NaN,1588,0.803,1.105,NaN
4,OAS1_0005_MR1,M,R,18,NaN,NaN,NaN,NaN,1737,0.848,1.010,NaN


In [4]:
# get the info of the data 

####### DR comments - delete when done#################################
# Be sure to do what is needed to change the types of the data if necessary
# There could be code clean up here - just document what you are doing and why

mri_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 436 entries, 0 to 435
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   ID      436 non-null    str    
 1   M/F     436 non-null    str    
 2   Hand    436 non-null    str    
 3   Age     436 non-null    int64  
 4   Educ    235 non-null    float64
 5   SES     216 non-null    float64
 6   MMSE    235 non-null    float64
 7   CDR     235 non-null    float64
 8   eTIV    436 non-null    int64  
 9   nWBV    436 non-null    float64
 10  ASF     436 non-null    float64
 11  Delay   20 non-null     float64
dtypes: float64(7), int64(2), str(3)
memory usage: 41.0 KB


### 2.1 Summary of Variables within the mri_df Table 

The mri_df table has 12 columns/variables and 436 rows.  All variables are explained in this section, including their descriptions, their units, their types, and some descriptive statistics.

#### A. CATEGORICAL VARIABLES

* **M/F** – M if is a male, F if is a female 
    * Number of missing values: 0
    * Contains 2 distinct values:
        * List the values.
        - M -> male
        - F -> female

* **Hand** – if the person is right or left handed
    * Number of missing values: 0
    * Contains 2 distinct values:
        * List the values.  
        - R -> right-handed
        - L -> left-handed

* **column_name** – description 
    * Number of missing values:
    * Units: if necessary; delete otherwise.
    * Contains XXX distinct values:
        * List the values.  If there are many, just list some to explain.  If there are only 10 or so, list them all and give a description of what they mean.  
    * Any other important facts, should you discover them during your analysis.

#### B. QUANTITATIVE  DISCRETE VARIABLES

* **Age** – the age of the person
    * Number of missing values: 0
    * Units: years
    * max value: 96
    * min value: 18
    * average value: 51.357798165137616

* **Educ** – Educational level
    * Number of missing values: 201
    * max value: 5
    * min value: 1
    * average value: 3.178723404255319

* **SES** – socioeconomic status
    * Number of missing values: 220
    * max value: 5
    * min value: 1
    * average value: 2.490740740740741

* **MMSE** – Mini Mental State Examination
    * Number of missing values: 201
    * max value: 30
    * min value: 14
    * average value: 27.06382978723404

* **CDR** – Clinical Dementia Rating
    * Number of missing values: 201
    * max value: 2
    * min value: 0
    * average value: 0.2851063829787234

* **eTIV** – Estimated Total Intracranial Volume
    * Number of missing values: 0
    * max value: 1992
    * min value: 1123
    * average value: 1481.9197247706422

* **nWBV** – Normalize Whole Brain Volume
    * Number of missing values: 0
    * max value: 0.893
    * min value: 0.644
    * average value: 0.7916697247706422

### 2.2 Code used to Summarize Variables

In this section we include the code used to describe the variables in the previous section.

In [36]:
#Showing the number of distinct values in each column to understand the data better
for col in mri_df.columns:
    print(f"{col}: {mri_df[col].nunique()} distinct values")


ID: 436 distinct values
M/F: 2 distinct values
Hand: 1 distinct values
Age: 73 distinct values
Educ: 5 distinct values
SES: 5 distinct values
MMSE: 17 distinct values
CDR: 4 distinct values
eTIV: 312 distinct values
nWBV: 182 distinct values
ASF: 282 distinct values
Delay: 14 distinct values


In [25]:
#Get the min, max, avergae, and number of missing values in the Age column
max = mri_df['Age'].max()
min = mri_df['Age'].min()
mean = mri_df['Age'].mean()

print(f"Max Age: {max}")
print(f"Min Age: {min}")
print(f"Mean Age: {mean}")

# Count null values in the 'Age' column
missing = mri_df['Age'].isna().sum()
print(f"Number of missing values in Age: {missing}")

Max Age: 96
Min Age: 18
Mean Age: 51.357798165137616
Número de valores nulos: 0


In [26]:
#Get the min, max, avergae, and number of missing values in the Educ column
max = mri_df['Educ'].max()
min = mri_df['Educ'].min()
mean = mri_df['Educ'].mean()

print(f"Max Educ: {max}")
print(f"Min Educ: {min}")
print(f"Mean Educ: {mean}")

# Count null values in the 'Educ' column
missing = mri_df['Educ'].isna().sum()
print(f"Number of missing values in Educ: {missing}")

Max Educ: 5.0
Min Educ: 1.0
Mean Educ: 3.178723404255319
Number of missing values in Educ: 201


In [28]:
#Get the min, max, avergae, and number of missing values in the SES column
max = mri_df['SES'].max()
min = mri_df['SES'].min()
mean = mri_df['SES'].mean()

print(f"Max SES: {max}")
print(f"Min SES: {min}")
print(f"Mean SES: {mean}")

# Count null values in the 'SES' column
missing = mri_df['SES'].isna().sum()
print(f"Number of missing values in SES: {missing}")

Max SES: 5.0
Min SES: 1.0
Mean SES: 2.490740740740741
Number of missing values in SES: 220


In [29]:
#Get the min, max, avergae, and number of missing values in the MMSE column
max = mri_df['MMSE'].max()
min = mri_df['MMSE'].min()
mean = mri_df['MMSE'].mean()

print(f"Max MMSE: {max}")
print(f"Min MMSE: {min}")
print(f"Mean MMSE: {mean}")

# Count null values in the 'MMSE' column
missing = mri_df['MMSE'].isna().sum()
print(f"Number of missing values in MMSE: {missing}")

Max MMSE: 30.0
Min MMSE: 14.0
Mean MMSE: 27.06382978723404
Number of missing values in MMSE: 201


In [30]:
#Get the min, max, avergae, and number of missing values in the CDR column
max = mri_df['CDR'].max()
min = mri_df['CDR'].min()
mean = mri_df['CDR'].mean()

print(f"Max CDR: {max}")
print(f"Min CDR: {min}")
print(f"Mean CDR: {mean}")

# Count null values in the 'CDR' column
missing = mri_df['CDR'].isna().sum()
print(f"Number of missing values in CDR: {missing}")

Max CDR: 2.0
Min CDR: 0.0
Mean CDR: 0.2851063829787234
Number of missing values in CDR: 201


In [31]:
#Get the min, max, avergae, and number of missing values in the eTIV column
max = mri_df['eTIV'].max()
min = mri_df['eTIV'].min()
mean = mri_df['eTIV'].mean()

print(f"Max eTIV: {max}")
print(f"Min eTIV: {min}")
print(f"Mean eTIV: {mean}")

# Count null values in the 'eTIV' column
missing = mri_df['eTIV'].isna().sum()
print(f"Number of missing values in eTIV: {missing}")

Max eTIV: 1992
Min eTIV: 1123
Mean eTIV: 1481.9197247706422
Number of missing values in eTIV: 0


In [32]:
#Get the min, max, avergae, and number of missing values in the nWBV column
max = mri_df['nWBV'].max()
min = mri_df['nWBV'].min()
mean = mri_df['nWBV'].mean()

print(f"Max nWBV: {max}")
print(f"Min nWBV: {min}")
print(f"Mean nWBV: {mean}")

# Count null values in the 'nWBV' column
missing = mri_df['nWBV'].isna().sum()
print(f"Number of missing values in nWBV: {missing}")

Max nWBV: 0.893
Min nWBV: 0.644
Mean nWBV: 0.7916697247706422
Number of missing values in nWBV: 0


---

## 3. Split into Train/Test

We now split the data into a training and a testing set for upcoming analysis.

In [8]:
#Set a seed for reproducibility
SEED = ... 

In [9]:
# split the data into training and testing
...

Ellipsis

In [10]:
# Unite the training data into a dataframe for the EDA
#replace xx with a good name and delete this code comment
xx_train_df = ...  

In [11]:
# Unite the testing data into a dataframe for the EDA
#replace xx with a good name and delete this code comment
xx_test_df = ...  

## 4. Initial EDA

**Students, in this section, ask some interesting questions about your (training) data in order to start learning about it.  Likely, the xxx_train_df dataframe will be the dataframe you user here.**

Prior to building any models, we begin our analysis by performing a exploratory data analysis on the **training data.**

We begin by considering ....


### 4.1 XXXX 

**As you learn about your data, create subsections with sensible titles to keep your analysis organized.  So what are you trying to learn about in this section? Include an introductory senrtence and a discussion after each graph/test/etc.  Remember that your eventual goal is flow and your file must be easily readable.**** 

### 4.2 XXXX

### 4.3 XXXX

etc.

### 4.XXX Conclusion/Summary

...

---

## 5. ML PLAYGROUND - Regression  (Will become part of the results section)


In this section, we use regression to ...

### 5.1 XXXX 

### 5.2 XXXX 

### 5.3 XXXX 

### 5.XXX Conclusion/Summary


---

## 6. ML PLAYGROUND - Classification (Will become part of the results section)

In this section, we use classification to ...

### 6.1 XXXX 

### 6.2 XXXX 

### 6.3 XXXX 

### 6.XXX Conclusion/Summary

---

## 7. Conclusion

Do not write this section until the final report is due at the end of the semester.  

When you are ready to write the conclusion:

* Summarize your results and what you’ve discovered.
* Indicate what related questions you’d ask in the future if you had time or if you could find the perfect data.
* I expect 2-3 paragraphs, more if needed.